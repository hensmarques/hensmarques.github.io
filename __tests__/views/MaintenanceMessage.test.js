import React from 'react';
import { MaintenanceMessage } from '../../src/views/MaintenanceMessage';

const DEFAULT_TEXT = 'Some pages are temporarily unavailable.';
const DEFAULT_TRANSLATE = '.CONTENT';
const DEFAULT_PAGE_TYPE = 'simple-ui';
const TRANSLATION = 'MAINTENANCE_MODE';

describe('<MaintenanceMessage/>', () => {
  const mockProps = {
    msg: {
      translate: DEFAULT_TRANSLATE,
      msg: DEFAULT_TEXT,
    },
    t: TRANSLATION,
    pageType: DEFAULT_PAGE_TYPE,
  };
  const wrapper = shallow(<MaintenanceMessage {...mockProps} />);

  it('renders 1 .maintenance-message inside', () => {
    expect(wrapper.find('.maintenance-message').length).toBe(1);
  });

  it('renders 1 .maintenance-message-content inside', () => {
    expect(wrapper.find('.maintenance-message-content').length).toBe(1);
    expect(wrapper.containsAllMatchingElements([
      <div
        className="maintenance-message-content"
        dangerouslySetInnerHTML={{
          __html: _t(mockProps.msg.msg, mockProps.msg.translate),
        }}
      />,
    ])).toBe(true);
  });

  it('renders and matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
