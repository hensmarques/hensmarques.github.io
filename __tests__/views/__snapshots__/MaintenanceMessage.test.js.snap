// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MaintenanceMessage/> renders and matches the snapshot 1`] = `
<div
  className="maintenance-message maintenance-message--simple-ui "
>
  <div
    className="maintenance-message-container maintenance-message-container--simple-ui"
  >
    <div
      className="maintenance-message-content"
      dangerouslySetInnerHTML={
        Object {
          "__html": "Some pages are temporarily unavailable.",
        }
      }
    />
    <div
      className="maintenance-message-close"
      onClick={[Function]}
    >
      <i
        className="fas fa-window-close"
      />
    </div>
  </div>
</div>
`;
